server_test.go

```diff
@@ -102,18 +107,18 @@ func RunLocalUDPServerUnsafe(laddr string) (*Server, string, error) {
 }
 
 func RunLocalTCPServer(laddr string) (*Server, string, error) {
-	server, l, _, err := RunLocalTCPServerWithFinChan(laddr)
+	server, l, _, err := RunLocalTCPServerWithFinChan(laddr, time.Hour, time.Hour)
 
 	return server, l, err
 }
 
-func RunLocalTCPServerWithFinChan(laddr string) (*Server, string, chan error, error) {
+func RunLocalTCPServerWithFinChan(laddr string, readTimeout time.Duration, writeTimeout time.Duration) (*Server, string, chan error, error) {
```

Comment:
This change is not needed you can set these values after the server is created, add shutdownTimeout will be calculated during shutdown.

Reply:
yes for shutdownTimeout, but the readTimeout will be used at readUDP() or readTCP() and that happen as soon as ServeAndActivate() is launched.